spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessorTests.java

```diff
@@ -273,6 +278,42 @@ public void queuesToDeclare() {
 		context.close();
 	}
 
+	@Test
+	public void concurrency() throws InterruptedException, ExecutionException {
+		final int concurrencyLevel = 8;
+		final ExecutorService executorService = Executors.newFixedThreadPool(concurrencyLevel);
+		try {
+			for (int i = 0; i < 1000; ++i) {
```

Comment:
@artembilan @garyrussell what do you think would be a better place for such a test?
Other tests here are take ~50ms, but this loop takes much longer. Is there a class with similar tests already, or should it be a new one?